src/Nethermind/Nethermind.Core/Caching/ICache.cs

```diff
@@ -8,8 +8,22 @@ public interface ICache<in TKey, TValue>
         void Clear();
         TValue? Get(TKey key);
         bool TryGet(TKey key, out TValue? value);
-        void Set(TKey key, TValue val);
-        void Delete(TKey key);
+
+        /// <summary>
+        /// Sets value in the cache.
+        /// </summary>
+        /// <param name="key"></param>
+        /// <param name="val"></param>
+        /// <returns>True if key didn't exist in the cache, otherwise false.</returns>
+        bool Set(TKey key, TValue val);
+
+        /// <summary>
+        /// Delete key from cache.
+        /// </summary>
+        /// <param name="key"></param>
+        /// <param name="val"></param>
+        /// <returns>True if key didn't exist in the cache, otherwise false.</returns>
+        bool Delete(TKey key);
```

Comment:
`True if key didn't exist in the cache, otherwise false.` - implementation is opposite. Should be `True if key existed in the cache, otherwise false.`

Reply:
fixed